app/src/main/java/com/xmartlabs/gong/domain/repository/LocationRepository.kt

```diff
@@ -1,10 +1,55 @@
 package com.xmartlabs.gong.domain.repository
 
+import com.xmartlabs.gong.data.model.Location
+import com.xmartlabs.gong.data.repository.location.LocationLocalSource
 import com.xmartlabs.gong.data.repository.location.LocationRemoteSource
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.merge
+import java.util.Date
+import kotlin.time.ExperimentalTime
+import kotlin.time.milliseconds
+import kotlin.time.minutes
 
 /**
  * Created by mirland on 28/04/20.
  */
-class LocationRepository(private val locationRemoteSource: LocationRemoteSource) {
-  suspend fun getLocation() = locationRemoteSource.getLocation()
+@OptIn(ExperimentalTime::class)
+class LocationRepository(
+    private val locationLocalSource: LocationLocalSource,
+    private val locationRemoteSource: LocationRemoteSource
+) {
+  companion object {
+    private val REFRESH_LOCATION_DURATION = 1.minutes
+  }
+
+  private var lastRequestedLocation: Date? = null
+
+  fun getLocation(): Flow<Location> {
+    return if (shouldFetchNewLocation()) {
+      listOf(getRemoteLocationFlow(), locationLocalSource.getLocation())
```

Comment:
shouldn't be flipped?  or merge will not take in count the order?

Reply:
the order doesn't matter, it's a merge, it's not a `concat`. I can do a `scan` and take the last one, but I don't want to add more complexity to the example. It works because when The item is added the db, the db will notify (it's a flow), so it's work fine.

Reply:
in that case i think that a merge its better